src/planner/plan/plan_join_order.cpp

```diff
@@ -265,7 +265,8 @@ void Planner::planNodeIDScan(uint32_t nodePos, const QueryGraphPlanningInfo& inf
     // later in the optimization stage. So we can assume the cardinality will not exceed outer
     // plan cardinality.
     if (plan->getCardinality() > info.corrExprsCard) {
-        plan->setCardinality(info.corrExprsCard);
+        plan->getLastOperator()->setCardinality(info.corrExprsCard);
+        cardinalityEstimator.addNodeIDDomOverride(*node->getInternalID(), info.corrExprsCard);
```

Comment:
I was seeing a problem where this updated cardinality (which I guess is from us assuming we'll add a semi-mask later) isn't in sync with the denominator in the cardinality estimator so I added a second map with overrides for the denominators (that is reset for each query graph). Is this the right way to deal with this?

Reply:
If it is re-initialized for each query graph, can we directly update existing nodeID dom?

Reply:
Discussed offline, renamed to `addPerQueryGraphNodeIDDom` to be more clear